onnx/utils.py

```diff
@@ -231,15 +232,22 @@ def extract_model(
     if check_model:
         onnx.checker.check_model(input_path)
 
-    model = onnx.load(input_path)
-
-    if infer_shapes:
-        model = onnx.shape_inference.infer_shapes(model)
+    if infer_shapes and os.path.getsize(input_path) > MAXIMUM_PROTOBUF:
+        onnx.shape_inference.infer_shapes_path(input_path, output_path)
+        model = onnx.load(output_path)
+    else:
+        model = onnx.load(input_path)
+        if infer_shapes:
+            model = onnx.shape_inference.infer_shapes(model)
 
     e = Extractor(model)
     extracted = e.extract_model(input_names, output_names)
 
-    onnx.save(extracted, output_path)
+    if extracted.ByteSize() > MAXIMUM_PROTOBUF:
+        location = os.path.basename(output_path) + "_data"
```

Comment:
```suggestion
        location = os.path.basename(output_path) + ".data"
```

consider a .data extension? When the model extension is not onnx, for example json, the original will produce json_data etc which is a little arbitrary

Reply:
Thank you @justinchuby , I changed it to ".data"